cmd/testmatrix/main.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This script generates the matrix.md file from most recent build of each
+// trigger.
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"html/template"
+	"log"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	e2e_testing "github.com/GoogleCloudPlatform/opentelemetry-operations-e2e-testing"
+	"github.com/alexflint/go-arg"
+	"google.golang.org/api/cloudbuild/v1"
+)
+
+const (
+	pass        status = ":white_check_mark:"
+	skip        status = ":leftwards_arrow_with_hook:"
+	templateTxt        = `# Matrix of supported scenarios in each ops repo
+
+<table>
+	<thead>
+		<tr>
+			<th>Repo Name</th>
+			<th>Platform</th>
+			{{- range $.Scenarios }}
+				<th>{{ . }}</th>
+			{{- end }}
+		</tr>
+	</thead>
+	<tbody>
+	{{- range $repoName := $.RepoNames }}
+		{{- range $i, $platform := $.Platforms }}
+			<tr>
+				{{- if eq $i 0 }}
+				<td rowspan={{ len $.Platforms }}>
+					<a href="https://github.com/GoogleCloudPlatform/{{ $repoName }}">{{ $repoName }}</a>
+				</td>
+				{{- end }}
+				<td>{{ $platform }}</td>
+				{{- range $scenario := $.Scenarios }}
+					<td>{{ index $.RepoToPlatformToScenario $repoName $platform $scenario }}</td>
+				{{- end }}
+			</tr>
+		{{- end }}
+	{{- end }}
+	</tbody>
+</table>
+
+- *{{ .Pass }} means passing*
+- *{{ .Skip }} means not implemented (skipped)*
+`
+)
+
+type Args struct {
+	e2e_testing.CmdWithProjectId
+}
+
+type status string
+
+type result struct {
+	RepoName string
+	Platform string
+	Statuses map[string]status
+}
+
+var (
+	triggerNameRe  = regexp.MustCompile(`^ops-\w+-e2e-.*$`)
+	scenarioPassRe = regexp.MustCompile(`: --- PASS:\s+([\w_]+)`)
+	scenarioSkipRe = regexp.MustCompile(`: --- SKIP:\s+([\w_]+)`)
+)
+
+func main() {
+	args := Args{}
+	arg.MustParse(&args)
+
+	ctx := context.Background()
+	cloudbuildService, err := cloudbuild.NewService(ctx)
+	if err != nil {
+		panic(err)
+	}
+	storageClient, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Is there a reason we aren't consistently calling either panic() or log.Fatal()?

Reply:
Switched to just use panic in main so defers will run if they are added